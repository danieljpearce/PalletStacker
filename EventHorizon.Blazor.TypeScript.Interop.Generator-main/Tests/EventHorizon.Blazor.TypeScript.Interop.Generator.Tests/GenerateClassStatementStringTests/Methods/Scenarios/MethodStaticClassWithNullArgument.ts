declare module Static.Method.Argument {
    export class ExampleClass {
        static undefinedArgument(arg1: null | number): boolean;
        static nullableArgument(arg1: Nullable<number>): boolean;
    }
}