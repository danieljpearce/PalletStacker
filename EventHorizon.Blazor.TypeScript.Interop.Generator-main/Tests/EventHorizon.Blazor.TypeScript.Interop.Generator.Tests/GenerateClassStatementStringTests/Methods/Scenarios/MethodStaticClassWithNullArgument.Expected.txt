/// Generated - Do Not Edit
namespace Static.Method.Argument
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods
        public static bool undefinedArgument(System.Nullable<decimal> arg1 = null)
        {
            return EventHorizonBlazorInterop.Func<bool>(
                new object[]
                {
                    new string[] { "Static", "Method", "Argument", "ExampleClass", "undefinedArgument" }, arg1
                }
            );
        }

        public static bool nullableArgument(System.Nullable<decimal> arg1 = null)
        {
            return EventHorizonBlazorInterop.Func<bool>(
                new object[]
                {
                    new string[] { "Static", "Method", "Argument", "ExampleClass", "nullableArgument" }, arg1
                }
            );
        }
        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}